kernel/tests/golden_tables.rs

```diff
@@ -0,0 +1,343 @@
+full_scan_test!("basic-with-inserts-updates");
+full_scan_test!("basic-with-vacuum-protocol-check-feature");
+skip_test!("checkpoint": "test not yet implmented");
+skip_test!("corrupted-last-checkpoint-kernel": "BUG: should fallback to old commits/checkpoint");
+skip_test!("data-reader-array-complex-objects": "list field expected name item but got name element");
+skip_test!("data-reader-array-primitives": "list field expected name item but got name element");
+full_scan_test!("data-reader-date-types-America");
+full_scan_test!("data-reader-date-types-Asia");
+full_scan_test!("data-reader-date-types-Etc");
+full_scan_test!("data-reader-date-types-Iceland");
+full_scan_test!("data-reader-date-types-Jst");
+full_scan_test!("data-reader-date-types-Pst");
+full_scan_test!("data-reader-date-types-utc");
+full_scan_test!("data-reader-escaped-chars");
+skip_test!("data-reader-map": "map field named 'entries' vs 'key_value'");
+full_scan_test!("data-reader-nested-struct");
+skip_test!("data-reader-nullable-field-invalid-schema-key":
+  "list field expected name item but got name element");
+skip_test!("data-reader-partition-values": "list field expected name item but got name element");
+full_scan_test!("data-reader-primitives");
+full_scan_test!("data-reader-timestamp_ntz");
+skip_test!("data-reader-timestamp_ntz-id-mode": "id column mapping mode not supported");
+full_scan_test!("data-reader-timestamp_ntz-name-mode");
+
+// TODO test with predicate
+full_scan_test!("data-skipping-basic-stats-all-types");
+full_scan_test!("data-skipping-basic-stats-all-types-checkpoint");
+skip_test!("data-skipping-basic-stats-all-types-columnmapping-id": "id column mapping mode not supported");
+full_scan_test!("data-skipping-basic-stats-all-types-columnmapping-name");
+full_scan_test!("data-skipping-change-stats-collected-across-versions");
+full_scan_test!("data-skipping-partition-and-data-column");
+
+full_scan_test!("decimal-various-scale-precision");
+
+// deltalog-getChanges test currently passing but needs to test the following:
+// assert(snapshotImpl.getLatestTransactionVersion(engine, "fakeAppId") === Optional.of(3L))
+// assert(!snapshotImpl.getLatestTransactionVersion(engine, "nonExistentAppId").isPresent)
+full_scan_test!("deltalog-getChanges");
+
+full_scan_test!("dv-partitioned-with-checkpoint");
+full_scan_test!("dv-with-columnmapping");
+skip_test!("hive": "test not yet implmented - different file structure");
+full_scan_test!("kernel-timestamp-int96");
+full_scan_test!("kernel-timestamp-pst");
+full_scan_test!("kernel-timestamp-timestamp_micros");
+full_scan_test!("kernel-timestamp-timestamp_millis");
+full_scan_test!("log-replay-dv-key-cases");
+full_scan_test!("log-replay-latest-metadata-protocol");
+full_scan_test!("log-replay-special-characters");
+full_scan_test!("log-replay-special-characters-a");
+full_scan_test!("multi-part-checkpoint");
+full_scan_test!("only-checkpoint-files");
+
+// TODO some of the parquet tests use projections
```

Comment:
we support projections, so why can't we do these?

Reply:
yea I think with your comment below I'll just try to unify all the tests under a `golden_table!` macro and we can just always pass in a 'test function' and I'll make one for full table scan, scan with predicate, etc. (or maybe a function that has that stuff as params

Reply:
taking as follow-up. I can make issues for some of these TODOs?